UnicornRender/video/include/unicorn/video/Camera.hpp

```diff
@@ -8,141 +8,172 @@
+
+    float m_aspect;
+    glm::mat4& m_cameraProjection;
+    system::Window* m_pWindow;
+    bool m_isDirty;
+};
+
+/** @brief Perspective camera projection controller */
+class PerspectiveCamera : public CameraProjection
+{
+public:
+    UNICORN_EXPORT PerspectiveCamera(system::Window* window, glm::mat4& cameraProj);
+
+    /** @brief Changes fov */
+    UNICORN_EXPORT void Scroll(float yoffset) override;
+
+    /** @brief Sets fov */
     UNICORN_EXPORT void SetFov(float fov);
 
-    /**
-     * @brief Returns direction
-     * @return Direction vector
-     */
-    UNICORN_EXPORT const glm::vec3& GetDirection() const;
-
-    /**
-     * @brief Returns up vector
-     * @return Up vector value
-     */
-    UNICORN_EXPORT const glm::vec3& GetUpVector() const;
-
-    /**
-     * @brief Returns projection matrix
-     * @return Projection matrix value
-     */
-    UNICORN_EXPORT const glm::mat4& GetProjection() const;
-
-    /**
-     * @brief Returns view matrix
-     * @return View matrix value
-     */
-    UNICORN_EXPORT const glm::mat4& GetView() const;
-
-    /**
-     * @brief Returns field of view value
-     * @return Field of view value
-     */
+    /** @brief Sets near Z plane */
+    UNICORN_EXPORT void SetZNear(float znear);
+
+    /** @brief Sets far Z plane */
+    UNICORN_EXPORT void SetZFar(float zfar);
+
+    /** @brief Returns fov */
     UNICORN_EXPORT float GetFov() const;
+private:
+    float m_fov, m_fovLowerBound, m_fovUpperBound, m_znear, m_zfar;
 
-    /**
-    * @brief Checks if view or projection need to be updated and update it, if needed
-    */
-    UNICORN_EXPORT void Frame();
+    void UpdateProjection() override final;
+};
+
+/** @brief Orthographic camera projection controller */
+class OrthographicCamera : public CameraProjection
+{
+public:
+    UNICORN_EXPORT OrthographicCamera(system::Window* window, glm::mat4& cameraProj);
+    
+    UNICORN_EXPORT void UpdateProjection() override final;
+    UNICORN_EXPORT void Scroll(float yoffset) override;
+    UNICORN_EXPORT void SetScale(float scale);
+
+    UNICORN_EXPORT float GetScale() const;
 private:
-    float m_aspect;
-    glm::vec3 m_position;
-    glm::vec3 m_upVector;
-    glm::vec3 m_direction;
-
-    struct
-    {
-        glm::mat4 m_projection;
-        glm::mat4 m_view;
-    } m_matrices;
-
-    float m_fov;
-    float m_znear, m_zfar;
-    bool m_dirtyView;
-    float m_fovLowerBound;
-    float m_fovUpperBound;
-    bool m_dirtyProjection;
-    /**
-     * @brief Calculates view matrix
-     */
-    void UpdateViewMatrix();
-    /**
-     * @brief Calculates projection matrix
-     */
-    void UpdateProjectionMatrix();
+    float m_orthoScale;
+    float m_scaledAspect;
+    float m_right, m_top;
+    float m_back, m_front;
 };
+
+/** @brief 2D camera controller */
+class Camera2DController : public CameraController
+{
+public:
+    UNICORN_EXPORT Camera2DController(glm::mat4& cameraView);
+
+    UNICORN_EXPORT void MoveUp(float deltaTime);
```

Comment:
Comments

Reply:
Do it really needs to write comments like
//Moves camera up
void MoveUp